libvast/src/validate.cpp

```diff
@@ -16,27 +16,41 @@ namespace vast {
 
 namespace {
 
-caf::error validate_(const vast::data& data, const vast::type& type,
-                     const enum vast::validate mode,
-                     const std::string& prefix = "", size_t depth = 0) {
+auto validate_opaque_(const vast::data& data) -> caf::error {
+  return caf::visit(vast::detail::overload{
+                      [](const record&) -> caf::error {
+                        return caf::error{};
+                      },
+                      [](auto const&) -> caf::error {
+                        return caf::make_error(ec::invalid_argument,
+                                               "only records may have 'opaque' "
+                                               "attribute");
```

Comment:
This essentially means that the "opaque" attribute may not be repurposed, which I guess is a good thing. But I suppose that means we need to document it as a reserved attribute somewhere.

Reply:
It's documented in `validate.hpp`. (and it's only 'reserved' with regards to this validate function, users should still be able to use freely)